javascript/modules/client/src/context.ts

```diff
@@ -71,12 +71,8 @@ export class Context {
       gasFeeEstimationFactor: Context.resolveGasFeeEstimationFactor(
         contextParams.gasFeeEstimationFactor,
       ),
-      ipfs: contextParams.ipfsNodes.map((config) =>
-        new IpfsClient(config.url, config.headers)
-      ),
-      graphql: contextParams.graphqlNodes.map((url) =>
-        new GraphQLClient(url)
-      ),
+      ipfs: this.useIpfs(contextParams.ipfsNodes),
+      graphql: this.useGraphql(contextParams.graphqlNodes),
```

Comment:
I'm a bit confused. I would have sworn that `useWeb3Providers` was basically a setter, and that we were using it to update the Context state. But apparently it is just a digest helper?
useIpfs and useGraphql feels odd too, I would have an internal `resolveIpfs`, `resolveWeb3Providers` and `resolveGraphql()` function doing this, and then `useXXX` doing the setting

Reply:
Ping

Reply:
see comment below